lib/sequelize.js

```diff
@@ -145,6 +149,7 @@ var Sequelize = function(database, username, password, options) {
     pool: {},
     quoteIdentifiers: true,
     hooks: {},
+    retry: {max: 5, match: ['SQLITE_BUSY: database is locked']},
```

Comment:
Well actually, if you add a global retry on stuff like `unable to serialize` then that will trigger for the COMMIT query and retry the commit where it should actually be retrying the entire transaction instead.

This could obviously be fixed by just setting this as default in `sequelize.transaction`, but how do we let users implement it if need be then?